api/v1/attributes.proto

```diff
@@ -0,0 +1,69 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package istio.mixer.v1;
+
+option go_package="istio.io/mixer/api/v1;mixerpb";
+
+// Contains a set of attributes related to an event.
+message Attributes {
+  // Attributes related to the source of an event.
+
+  uint32 source_ip = 1;
+  uint32 source_port = 2;
+  string source_name = 3;
+  string source_instance = 4;
+  string source_namespace = 5;
+  string source_user = 6;
+  // ...
+
+  // Attributes related to the target of an event.
+  uint32 target_ip = 101;
+  uint32 target_port = 102;
+  string target_service = 103;
+  string target_name = 104;
+  string target_instance = 105;
+  string target_namespace = 106;
+  string target_user = 107;
+  // ...
+
+  // Attributes related to the origin of an event,
+  // if the source of the event is proxying it.
+  uint32 origin_ip = 201;
+  uint32 origin_port = 202;
+  string origin_name = 203;
+  string origin_instance = 204;
+  string origin_namespace = 205;
+  string origin_user = 206;
+  // ...
+
```

Comment:
Eventually absolutely, I intentionally did not go whole hog so we could add them piece at a time :)

We can always start using them ahead of time without this since it is just an optimization.